ethereumj-core/src/main/java/org/ethereum/sync/SyncQueueImpl.java

```diff
@@ -244,16 +271,30 @@ private boolean addHeaderPriv(BlockHeaderWrapper header) {
 
     @Override
     public synchronized HeadersRequest requestHeaders() {
+        return requestHeadersImpl(MAX_CHAIN_LEN);
+    }
+
+    @Override
+    public synchronized HeadersRequest requestHeaders(int count) {
+        return requestHeadersImpl(count);
+    }
+
+    private HeadersRequest requestHeadersImpl(int count) {
+        long startNumber;
+        boolean reverse = false;
         if (!hasGaps()) {
-            return new HeadersRequestImpl(maxNum + 1, MAX_CHAIN_LEN, false);
+            startNumber = maxNum + 1;
         } else {
             List<HeaderElement> longestChain = getLongestChain();
-            if (rnd.nextBoolean()) {
-                return new HeadersRequestImpl(longestChain.get(longestChain.size() - 1).header.getNumber(), MAX_CHAIN_LEN, false);
-            } else {
-                return new HeadersRequestImpl(longestChain.get(longestChain.size() - 1).header.getNumber(), MAX_CHAIN_LEN, true);
-            }
+            startNumber = longestChain.get(longestChain.size() - 1).header.getNumber();
+            if (!rnd.nextBoolean()) reverse =  true;
```

Comment:
When requesting reverse headers to close a gap no need to request `count` headers it makes sense to request only `MAX_CHAIN_LEN`